libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -3020,15 +3042,18 @@ narrowByFuelAndSize fuel size specs =
       , 0 `conformsToSpec` elemS =
           Just (elemS <> notEqualSpec 0, foldS)
       -- Member specs with non-zero elements, TODO: explain
-      | MemberSpec xs <- elemS
+      | MemberSpec ys <- elemS
+      , let xs = NE.toList ys
       , Just u <- knownUpperBound foldS
       , all (0 <=) xs
       , any (0 <) xs
       , let xMinP = minimum $ filter (0 <) xs
             possible x = x == u || xMinP <= u - x
             xs' = filter possible xs
       , xs' /= xs =
-          Just (MemberSpec $ nubOrd xs', foldS)
+          case NE.nonEmpty (nubOrd xs') of
+            Nothing -> Nothing -- IS THIS RIGHT?
+            Just ys -> Just (MemberSpec ys, foldS)
```

Comment:
This doesn't preserve the old semantics. Better to use your utility function.

Reply:
How about this
```
 xs' /= xs = Just (memberSpecList (nubOrd xs') (pure ("None of "++show xs++" are possible")), foldS)
```